typed/parser.go

```diff
@@ -105,10 +107,41 @@ func (p *ParseableType) FromYAML(object YAMLObject) (TypedValue, error) {
 
 // FromUnstructured converts a go interface to a TypedValue. It will return an
 // error if the resulting object fails schema validation.
-func (p *ParseableType) FromUnstructured(in interface{}) (TypedValue, error) {
+func (p *parseableType) FromUnstructured(in interface{}) (TypedValue, error) {
 	v, err := value.FromUnstructured(in)
 	if err != nil {
 		return TypedValue{}, err
 	}
 	return AsTyped(v, &p.parser.Schema, p.typename)
 }
+
+// DeducedParseableType is a ParseableType that deduces the type from
+// the content of the object.
+type DeducedParseableType struct{}
```

Comment:
Why is DeducedParseableType exported, but parsableType isn't?

Reply:
Is it because this type is just a struct{} alias?

Reply:
Because the other one needs to be created in a "fancy way", so it's build through a method that returns the interface?
I could do
```
func NewDeducedParseableType() ParseableType {
    return deducedParseableType{}
}
```
but I don't think it's very useful

Reply:
in fewer words, yes =)